ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -52,6 +53,89 @@ def __init__(self):
         }
 
 
+class CoffeeDemoTask():
+    """
+    Representation of the coffee demo recipe defined
+    by its steps and transitions between them.
+
+    NOTE: This currently only represents first 13 steps of the coffee demo.
+    Steps are listed under v1.2 here:
+    https://docs.google.com/document/d/1MfbZdRS6tOGzqNSN-_22Xwmq-huk5_WdKeDSEulFOL0/edit#heading=h.l369fku95vnn
+    """
+    def __init__(self):
+        self.name = 'Pour-over coffee'
+
+        self.items = {'scale': 1, '25g coffee beans': 1, 'mug': 1, '12 oz water': 1,
+                      'kettle': 1, 'grinder': 1, 'measuring cup': 1, 'bowl': 1}
+
+        self.description = ('Boil water, grind coffee beans, and' +
+                            ' place the coffee filter into the dripper,'
+                            ' and then place the dripper on top of the mug')
+
+        # TODO: make a labels.txt file for these steps
+        self.steps = [
+            {'name': 'Pour_12oz_of_water_into_a_liquid_measuring_cup'},
+            {'name': 'Pour_water_from_the_liquid_measuring_cup_into_the_electric_kettle'},
+            {'name': 'Turn_on_the_kettle'},
+            {'name': 'Turn_on_kitchen_scale'},
+            {'name': 'Place_bowl_on_scale'},
+            {'name': 'Zero_scale'},
+            {'name': 'Add_coffee_beans_until_scale_reads_25_grams'},
+            {'name': 'Pour_coffee_beans_into_coffee_grinder'},
+            {'name': 'Take_the_coffee_filter_and_fold_it_in_half_to_create_a_semi_circle'},
+            {'name': 'Fold_the_filter_in_half_again_to_create_a_quarter_circle'},
+            {'name': 'Place_the_folded_filter_into_the_dripper_such_that_the_point' +
+                     '_of_the_quarter_circle_rests_in_the_center_of_the_dripper'},
+            {'name': 'Spread_the_filter_open_to_create_a_cone_inside_the_dripper'},
+            {'name': 'Place_the_dripper_on_top_of_the_mug'},
+            {'name': 'Done'}
+        ]
+
+        # Create the transitions between steps, assuming linear steps
+        self.transitions = []
+        for idx, step in enumerate(self.steps):
+            # No transition needed for the last step
+            if idx < (len(self.steps) - 1):
+                self.transitions.append({'trigger': step['name'],
+                                         'source': step['name'],
+                                         'dest': self.steps[idx + 1]['name']})
+
+        self.machine = Machine(model=self, states=self.steps,
```

Comment:
It took me a second to understand that this was modifying `self` with additional attributes and functions. Is there a quick link to the transitions docs about this for reference?

Reply:
I added a link in the class doc string to creating the state machines. Technically, `steps` and `transitions` don't have to be `self` attributes, but I use them later to figure out what the next activity should be and what the list of steps is.